physics/clm_lake.f90

```diff
@@ -0,0 +1,5737 @@
+
+    !
+    ! Configuration and initialization:
+    !
+    INTEGER, INTENT(IN) :: iopt_lake, iopt_lake_clm
+    REAL(KIND_PHYS), INTENT(IN)  :: min_lakeice, lakedepth_default, dtp
+    LOGICAL, INTENT(IN) :: use_lakedepth
+    INTEGER, DIMENSION(:), INTENT(IN) :: use_lake_model
+    REAL(KIND_PHYS), INTENT(INOUT) :: clm_lake_initialized(:)
+    LOGICAL, INTENT(IN) :: frac_grid, frac_ice
+
+    !
+    ! Atmospheric model state inputs:
+    !
+    REAL(KIND_PHYS), DIMENSION(:), INTENT(IN):: &
+         tg3, pgr, zlvl, qvcurr, xlat_d, xlon_d, ch, cm, &
+         dlwsfci, dswsfci, emiss, oro_lakedepth, wind, rho0, tsfc, &
+         rainncprv, raincprv
+    REAL(KIND_PHYS), DIMENSION(:,:), INTENT(in) :: gu0, gv0, prsi, gt0, phii
+    LOGICAL, DIMENSION(:), INTENT(IN) :: flag_iter
+    INTEGER, DIMENSION(:), INTENT(IN) :: ISLTYP
+
+    !
+    ! Feedback to atmosphere:
+    !
+    REAL(KIND_PHYS), DIMENSION(:), INTENT(INOUT) :: &
+         evap_wat,     evap_ice,   hflx_wat,    hflx_ice,  gflx_wat, gflx_ice,    &
+         ep1d_water,   ep1d_ice,   tsurf_water, tsurf_ice, tsfc_wat, tisfc,       &
+         weasdi,       snodi,      hice,        qss_water, qss_ice,               &
+         cmm_water,    cmm_ice,    chh_water,   chh_ice,                          &
+         uustar_water, uustar_ice, lake_t_snow, albedo,    zorlw,                 &
+         zorli,        lake_t2m,   lake_q2m,    weasd,     snowd,    fice
+    LOGICAL, INTENT(INOUT) :: icy(:)
+
+    !
+    ! Lake model internal state stored by caller:
+    !
+    INTEGER, DIMENSION( : ), INTENT(INOUT)    :: salty
+
+    real(kind_phys),           dimension(: )                ,intent(inout)  :: savedtke12d,    &
+                                                                               snowdp2d,       &    
+                                                                               h2osno2d,       &    
+                                                                               snl2d,          &    
+                                                                               t_grnd2d
+    
+    real(kind_phys),    dimension( :,: )           ,INTENT(inout)  :: t_lake3d,       &    
+                                                                                  lake_icefrac3d
+    real(kind_phys),    dimension( :,-nlevsnow+1: )  ,INTENT(inout)  :: t_soisno3d,     &    
+                                                                                  h2osoi_ice3d,   &    
+                                                                                  h2osoi_liq3d,   &    
+                                                                                  h2osoi_vol3d,   &    
+                                                                                  z3d,            &    
+                                                                                  dz3d 
+    real(kind_phys),    dimension( :,-nlevsnow+0: )  ,INTENT(inout)  :: zi3d    
```

Comment:
You are mistaken. This is a differed shape array since there is no upper bound. The variable has to have a lower bound for the code to work. Specifying a lower bound does not make it cease to be a deferred shape array.

Reply:
see prior comment